# Enterotool

This pipeline employs Seqkit for initial quality control (QC1) to ensure the integrity of the input data. Following this, the reads are processed using Trimmomatic to perform quality trimming, removing low-quality bases and sequences that do not meet the minimum read length or quality score thresholds. The trimmed reads undergo a second quality control step (QC2) to validate the effectiveness of the trimming process. The forward and reverse reads are then merged using Vsearch, followed by identifying and removing potential chimeric sequences via a de novo approach. The processed reads are aligned against a curated database using Blastn, employing stringent thresholds (e.g., 98% nucleotide identity) to ensure accurate genogroup identification. An in-house bash script parses the BLAST results, counting unique and valid matches for each genogroup. The resulting data is normalized to Counts Per Million (CPM) for sample consistency. Finally, the normalized data is visualized using Matplotlib, producing a genus-level stacked bar chart that highlights the relative abundances of identified genogroups. The process concludes with the generation of a comprehensive report summarizing the findings, ensuring reproducibility and interpretability of the results.

The workflow was designed for rapid and efficient sequencing data processing, ensuring specific identification of EV genogroups. All analyses were performed on a Linux-based system (Ubuntu 22.04.5 LTS) to optimize computational performance and reproducibility.

![Sin t√≠tulo](https://github.com/user-attachments/assets/5094931f-be2f-4f83-aaf9-fca1707bd68f)

# Dependencies 
The following tools and libraries are required to run the pipeline:
Bash.
Trimmomatic (v0.39).
Seqkit.
Vsearch (v2.29.1).
Blastn (NCBI BLAST+).
pandas.
matplotlib.
This database was obtained from the NCBI Virus repository (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/), and manually curated.

# References

NCBI Virus repository: https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/
Seqkit: Wei Shen et al., 2024.
Trimmomatic: Bolger et al., 2014.
vsearch: Rognes et al., 2016.
Blastn: Altschul et al., 1990.
Matplotlib: J. D. Hunter, "Matplotlib: A 2D Graphics Environment", Computing in Science & Engineering, vol. 9, no. 3, pp. 90-95, 2007.
