# Enterotool
Bioinformatic analyses were performed to identify the EV genogroups. A comprehensive database was constructed that included the full genomic variability of the target group obtained from the NCBI Virus repository (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/). Raw sequencing reads were processed and quality-trimmed using Seqkit (Wei Shen et al., 2024). Additional filtering, adapter checking, and chimera assessment were performed using Trimmomatic and VSEARCH's ensuring the exclusion of potential chimeric sequences from the dataset (Bolger et al., 2014; Rognes et al., 2016).  Genogroup identification and annotation were performed using the Blastn (Altschul et al., 1990). Data normalization was performed using the RNAnorm package (v2.1.0) (Zmrzlikar et al., 2023). An in-house bash script was developed for automated quantification of genogroup abundance, parsing Blastn output files to count the different matches identified per sample. Data visualization was conducted using matplotlib (v3.1.1) (Caswell et al., 2019)
