# Enterotool
A comprehensive database was constructed, incorporating the full genomic variability of the target group, obtained from the NCBI Virus repository (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/). Raw sequencing reads were processed using Seqkit for quality trimming (Shen et al., 2024). Additional filtering and adapter checking were performed using Trimmomatic (Bolger et al., 2014). The identification and annotation of serogroups were performed using the Blastn(Altschul et al., 1990). Data normalization was performed using the rnanorm package (v2.1.0) (Znrzlikar et al., 2023). An in-house bash script was developed for automated quantification of serogroup abundance, parsing Blastn output files to count the different matches identified per sample. Data visualization was conducted using matplotlib (v3.1.1) (Caswell et al., 2019)
